sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuUserDefinedFunction.scala

```diff
@@ -75,6 +75,17 @@ object GpuUserDefinedFunction {
   // UDFs can support all types except UDT which does not have a clear columnar representation.
   val udfTypeSig: TypeSig = (TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL +
       TypeSig.BINARY + TypeSig.CALENDAR + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT).nested()
+
+  /**
+   * Detect whether the assertion in a JVM where the method is called is enabled.
+   * (Yes, this is an ugly solution, but I fail to figure out a better way.)
+   */
+  def isAssertionEnabled: Boolean = try {
+    assert(false)
```

Comment:
I am not 100% sure how scala asserts translate to java asserts.  Also java asserts can be controlled per package, so this would not detect them if someone just enabled asserts for `ai.rapids.cudf`. I would almost rather just assume assertions are enabled unless we have a better way to detect them.

Reply:
Scala asserts are not translated to Java asserts.  IIUC Scala asserts always fire unless explicitly disabled during compilation.

Reply:
Really appreciate all of this. Good to know about Java and Scala assertion. Here we should concern only the Java assertion as you know.

And I just found the API `public boolean desiredAssertionStatus()` in `java.lang.Class` matches our requirements here.